packages/core/src/codewhisperer/service/testGenHandler.ts

```diff
@@ -94,11 +94,13 @@ export async function createTestJob(
     logger.debug('target line range end: %O', firstTargetLineRangeList?.end)
 
     const resp = await codewhispererClient.codeWhispererClient.startTestGeneration(req).catch((err) => {
+        ChatSessionManager.Instance.getSession().startTestGenerationRequestId = err.requestId
```

Comment:
so `ChatSessionManager.Instance.getSession()` is global, shared, mutable state?

Reply:
Yes, its global and mutable.

Reply:
It's clear that will be a source of bugs, right?